### DISCUSSION TITLE
Azure registered public server works Private does not

### DISCUSSION THREAD

**Discussion by ikelleyTCF** [jamulussoftware/jamulus#2967] [Technical Support & Help]
My client can connect when I make a Jamulus server a registered public serve.
When it is set to private, I can't connect. 

The server is running on an azure VM running windows.

**Comment by pljones** [comment]
Have you read the manual about running a private server?  Have you set up port forwarding to enable inbound unsolicited UDP requests to reach your server?

**Comment by ikelleyTCF** [comment]
I did say the registered worked. So UDP is working.
I did say it was an azure machine,. SO no need for port forwarding.

Would you like a manual on reading comprehension?

**Reply by pljones** [reply]
Hello.  Don't be rude if you want help.

The point of registering is two-fold.  It enables UDP hole punching as well as making your server visible.  The UDP hole punch avoids the need to configure port forwarding locally.

Please read the manual.

**Reply by henkdegroot** [reply]
> I did say it was an azure machine,. SO no need for port forwarding.

You will need to ensure the port is forwarded/open for sure. By default it is not on azure. My private server works fine on azure. Not using windows though, I am using linux.

**Reply by ikelleyTCF** [reply]
Thank you. 
So, I read the manual (5th time)

1. VM is wide open firewall, no ports blocked for UDP.
2. It is on an Azure VM not behind a home router, so I don't think I need to do that step from the "Manual".
3. cobbled this together based on "Manual". it is the public ip.

       --serverbindip XX.XXX.XXX.XXX --port 200 --server

Ran it...
C:\Program Files\Jamulus>jamulus --serverbindip XX.XXX.XXX.XXX --port 200 --server
- server bind IP: XX.XXX.XXX.XXX

- selected port number: 200
- server mode chosen


Got Error Message
Error Message: Network Error : Connot bind the socket (maybe the software is already running).

It is not running.

**Reply by ikelleyTCF** [reply]
OK. I think I know what is going on even though I can ping the VM by external IP the VM does not see it. I think Azure requires I use a static Ip for the Vm 'll try it.

**Reply by henkdegroot** [reply]
To my knowledge, the default behaviour of the Azure network is to block all incoming traffic and as jamulus is not initiating to make the "outbound" connection the port will not be "opened". If the external IP address is dynamic, that should not make any difference. That just means the client has to connect to a different ip every time. 

Maybe some information here that helps you: [https://thejunkyard.cc/jekyll/update/2021/03/05/Jamulus-Azure.html](https://thejunkyard.cc/jekyll/update/2021/03/05/Jamulus-Azure.html)
Even though it references Linux, the network part in Azure would be similar to setup.

**Reply by ikelleyTCF** [reply]
Thank I did those steps with windows VM. Those directions work because it is still a registered server. The last line reads 
"Note: Your server may take a moment to appear in the Jamulus server list". Even though it is one he made in Azure, other people can still join the server. I am looking into networking with NAT stuff in azure. Somewhere in the IPv6 to IPv4 mix is the answer. Just never needed to do it.

**Reply by ikelleyTCF** [reply]
> > I did say it was an azure machine,. SO no need for port forwarding.
> 
> You will need to ensure the port is forwarded/open for sure. By default it is not on azure. My private server works fine on azure. Not using windows though, I am using linux.

Did you just use the open udp port at the fire wall rules blade? Or did you need to setup a whole private network?

**Reply by henkdegroot** [reply]
I just went into Network Security Group and on the Inbound Rules tab, and added an Inbound Rule for Jamulus. I am using IPv4.

![image](https://user-images.githubusercontent.com/13550012/204055889-eec8d5bc-79bb-4b4f-a998-d17ea05296f1.png)

**Reply by ikelleyTCF** [reply]
I think i'm getting closer. Where do you specify ipv4? I have that setting and still no luck I also added a second ip config and found a toggle that enable port forwarding on the machine. Is there a place in the rules

**Reply by ikelleyTCF** [reply]
sorry it is ip forwarding not port on the toggle

**Reply by henkdegroot** [reply]
I don't remember having to specify it, but might be that when I setup my machine it was by default using IPv4 and maybe azure is using IPv6 as default now. 
For the network interface in azure, I have the following:
![image](https://user-images.githubusercontent.com/13550012/204076399-02fe58fe-33d1-4787-86c3-15316979fc0d.png)

For the virtual network in azure, I have the following:
![image](https://user-images.githubusercontent.com/13550012/204076486-cd8b7906-8fbf-4d58-91d6-9714f431f481.png)

**Reply by henkdegroot** [reply]
@ezoc71 I also note you wrote something about using -serverbindip option. I don't believe you will need that when you just setup a single private/not-registered server and I am not sure on the windows OS if the firewall is enabled, if it is you may need to allow an inbound rule on that as well for the port.

**Reply by ikelleyTCF** [reply]
Thank you I'm looking at that right now. Thanks for the screen shots. I have the exact same settings. I think this will be an OS/Machine thing.
I think all of the external config for the VM is good. Thank you for the help.

**Reply by ikelleyTCF** [reply]
I don't know what it means, but in the manual, there is a headless install for linux but not for Windows.
[latest headless (amd64) .deb file](https://github.com/jamulussoftware/jamulus/releases/download/r3_9_1/jamulus-headless_3.9.1_ubuntu_amd64.deb)

I'm going to switch to trying it with linux there are mor examples online.

**Reply by henkdegroot** [reply]
I believe a windows headless version might be made available in the future, as so far there is not much need for it.
Headless means it does not contain a GUI and does not have audio support. So the size will be much smaller.

Not sure why you could not get the windows version to work. Sounds like you are close....

**Reply by dcorson-ticino-com** [reply]
I had a private Win10 server running on Asure some time ago without problems.
As I remember, however, there was some work to do opening ports. 
I stopped at the end of the evaluation period as the ping times weren't that good.
